### PR TITLE
Issue 919: Upgrade Spring Boot to 2.1.0 version

### LINKED ISSUE #919: Upgrade `org.springframework.security:spring-security-ldap` to versions `5.0.9`, `5.1.1`, or higher
# Overview
This issue has been reported by snyk.io.
[org.springframework.security:spring-security-ldap](https://springframework.org/) provides security services for the Spring IO Platform. 

Affected versions of the package are vulnerable to Authentication Bypass.

# Remediation
Upgrade `org.springframework.security:spring-security-ldap` to versions `5.0.9`, `5.1.1`, or higher.

# References
[GitHub Commit](https://github.com/spring-projects/spring-security/commit/7269aacbdd42183afeedc84e2233580b917da4c0)
[GitHub Issue](https://github.com/spring-projects/spring-security/issues/3392)

### PR DESCRIPTION
**Pull request by ptirador**
Fixes #919.

This version [disables bean overriding](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.1-Release-Notes#bean-overriding) by default, so setting `spring.main.allow-bean-definition-overriding` to `true` is needed.

### CODE REVIEW

**Comment by carlspring** [comment]
@fuss86 / @sbespalov ;

Could you please review this? Thanks! :)

File: strongbox-web-core/src/main/resources/application.properties [hunk 1]
```diff
@@ -1,5 +1,6 @@
 server.port=48080
 spring.liquibase.enabled=false
+spring.main.allow-bean-definition-overriding=true
```
Comment (sbespalov) [inline strongbox-web-core/src/main/resources/application.properties]: @ptirador why we need this property?

Comment (steve-todorov) [reply strongbox-web-core/src/main/resources/application.properties]: @sbespalov it's mentioned in the PR's description :) 

> This version [disables bean overriding](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.1-Release-Notes#bean-overriding) by default, so setting `spring.main.allow-bean-definition-overriding` to `true` is needed.

Comment (sbespalov) [reply strongbox-web-core/src/main/resources/application.properties]: ok, I will ask another way :)
why do we need to enable bean definition overriding?

Comment (ptirador) [reply strongbox-web-core/src/main/resources/application.properties]: If we don't enable this, the following error appears when executing integration tests:

APPLICATION FAILED TO START
***************************

Description:

The bean 'authenticationManager', defined in class path resource [org/carlspring/strongbox/configuration/StrongboxSecurityConfig.class], could not be registered. A bean with that name has already been defined in URL [jar:file:/home/nullptr/.m2/repository/org/carlspring/strongbox/strongbox-security-api/1.0-SNAPSHOT/strongbox-security-api-1.0-SNAPSHOT.jar!/org/carlspring/strongbox/security/managers/AuthenticationManager.class] and overriding is disabled.

Action:

Consider renaming one of the beans or enabling overriding by setting spring.main.allow-bean-definition-overriding=true

Comment (fuss86) [reply strongbox-web-core/src/main/resources/application.properties]: ok 👍

Comment (sbespalov) [reply strongbox-web-core/src/main/resources/application.properties]: @ptirador can we resolve naming issue instead?

Comment (fuss86) [reply strongbox-web-core/src/main/resources/application.properties]: I just realized that `org.carlspring.strongbox.security.managers.AuthenticationManager` is probably initialized twice. It is marked as `@Component` so it takes part in component scanning and next we try to initialize it explicitly in `org.carlspring.strongbox.configuration.StrongboxSecurityConfig#authenticationManager`.

This is a bug. It needs to be solved.

Comment (carlspring) [reply strongbox-web-core/src/main/resources/application.properties]: I agree. @ptirador : Would you mind investigating this?

Comment (ptirador) [reply strongbox-web-core/src/main/resources/application.properties]: I'm trying to solve this but I'm not able to. Could you help me, @fuss86?

Comment (fuss86) [reply strongbox-web-core/src/main/resources/application.properties]: Have you tried erasing 

```
@Component
@Scope("singleton")
```
from `org.carlspring.strongbox.security.managers.AuthenticationManager` ?

Comment (ptirador) [reply strongbox-web-core/src/main/resources/application.properties]: Thank you so much! I've solved that and a few more bean issues related.

Comment (fuss86) [reply strongbox-web-core/src/main/resources/application.properties]: Glad I could help :)

**Comment by carlspring** [comment]
Thanks for your help! :)

**Pull request merged**